### DISCUSSION TITLE
Integrating SOFA into website or app

### DISCUSSION THREAD

**Discussion by giuliadinardo1** [sofa-framework/sofa#5083] [Ideas / Suggestions]
I was wondering if it is possible to add SOFA into a website or app so that other people can access scenes that use certain plugins rather than having to download and set up sofa themselves.

**Comment by alxbilger** [comment]
Plugins are precisely what makes it difficult to compile SOFA for the web. Currently, no efforts are made to do what you suggest. But it would be great, I agree. 

Have you some knowledge on the transfert of a C++ program on the web?

**Comment by giuliadinardo1** [comment]
Hi Alex, thank you for the response! 
I do not have much experience with C++. Do you have any suggestions on how to go about doing this? Would it be easier to create an app instead of a webpage?

**Reply by alxbilger** [reply]
There is already an application. It's `runSofa` (https://sofa-framework.github.io/doc/using-sofa/runsofa/).

To embed a C++ program in a webpage, https://emscripten.org/ is the tool to use. But as I said, the plugins system is a difficulty. There are also design issues to compile a static version of SOFA (but some works are progress to change that).

**Reply by giuliadinardo1** [reply]
Thank you for the resources! Is it possible to alter runsofa with all the compilation (with specific plugins) already complete so others can easily download it and see specific scenes?

**Comment by damienmarchal** [comment]
I always had dream of having sofa in a website. If some of you makes progress I would be super happy to hear about that.
